k4GeneratorsConfig/src/pythiaLHERunner.cxx

```diff
@@ -1,89 +1,120 @@
+              << "      Input LHE events file: " << lheFilePath << "\n"
+              << "      HepMC3 output filepath: " << hepmcFilePath
+              << std::endl;
+  }
 
-  //read the options
-  std::string filename="Pythia.dat";
-  std::string lhefilename="Pythia.lhe";
-  std::string hepmcFile="Pythia.hepmc";
-  int c;
-  while ((c = getopt (argc, argv, "f:l:o:")) != -1)
-    switch (c){
-    case 'f':
-      filename = optarg;
-      break;
-    case 'l':
-      lhefilename = optarg;
-      break;
-    case 'o':
-      hepmcFile = optarg;
-      break;
-    case 'h':
-      std::cout << "Usage: pythiaLHERunner -h -f filename -l filename" << std::endl;
-      std::cout << "-h: print this help" << std::endl;
-      std::cout << "-f filename: input file containing the pythia commands" << std::endl;
-      std::cout << "-l filename: input file containing the LHE events" << std::endl;
-      std::cout << "-o filename: output file containing the hepmc events" << std::endl;
-      exit(0);
-    default:
-      exit(0);
+  // Check if the Pythia file can be read
+  {
+    std::ifstream infile(pythiaCmdFilePath);
+    if (!infile.good()) {
+      std::cout << "pythiaLHErunner::ERROR: Input Pythia command file with "
+                << "name \""
+                << pythiaCmdFilePath << "\" cannot be read!\n"
+                << "                        Aborting..." << std::endl;
+      exit(1);
     }
-  // check existence of the file:
-  std::ifstream infile(filename);
-  if ( infile.fail() ) {
-    std::cout << "pythiaLHERunner:: input file with name " << filename << " not found. Exiting" << std::endl;
-    exit(0);
+    infile.close();
   }
 
-  // check existence of the file:
-  std::ifstream inLHEfile(filename);
-  if ( inLHEfile.fail() ) {
-    std::cout << "pythiaLHERunner:: input file with name " << lhefilename << " not found. Exiting" << std::endl;
-    exit(0);
+  // Check if the input LHE file can be read
+  {
+    std::ifstream inLHEfile(lheFilePath);
+    if (!inLHEfile.good()) {
+      std::cout << "pythiaLHErunner::ERROR: Input LHE file with name \""
+                << lheFilePath
+                << "\" cannot be read!.\n"
+                << "                        Aborting..." << std::endl;
+      exit(1);
+    }
+    inLHEfile.close();
   }
-  // Pythia generator.
-  Pythia pythia;
-  // add the write hepmc flag to the settings
-  pythia.settings.addFlag("Main:writeHepMC",false);
+
+  // Pythia generator
+  Pythia8::Pythia pythia;
+  // Add the write hepmc flag to the settings
+  pythia.settings.addFlag("Main:writeHepMC", false);
   // Read in the rest of the settings and data from a separate file.
-  pythia.readFile(filename);
+  pythia.readFile(pythiaCmdFilePath);
 
-  // check for hepmc
+  // Check for HepMC
   const bool hepmc = pythia.flag("Main:writeHepMC");
-  std::cout << "HEPMC file name is " << hepmcFile << std::endl;
+  const std::string hepmcFilePathFromCmdFile = pythia.word("Main:HepMCFile");
```

Comment:
Keeping output HepMC file specified in the command file for reference later (if exists)